### PR TITLE
Use more functionality from golang.org/x/sys/unix

### PR DESCRIPTION
**Pull request by tklauser**
**- What I did**

Updated the vendored copy of the `golang.org/x/sys` package. Changed `pkg/loopback`, `pkg/system` and `pkg/dmesg?  to use functions, types and constants from `golang.org/x/sys/unix` instead of duplicating them.

**- How I did it**

Replaced manual re-implementations of said functions, types and constants with equivalent counterparts provided by `golang.org/x/sys/unix`.

**- How to verify it**

Compare current implementation to respective counterparts in `x/sys/unix`.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

Amazona collaria

### CODE REVIEW

**Comment by thaJeztah** [comment]
😍 thanks!

**Comment by thaJeztah** [comment]
@kolyshkin ptal

**Comment by thaJeztah** [comment]
@tonistiigi @kolyshkin ptal

**Review by tonistiigi — APPROVED** [review]
@tklauser please rebase

**Comment by thaJeztah** [comment]
vendoring was updated in https://github.com/moby/moby/issues/39726, which should have all the changes needed for this PR, so the vendor bump can be dropped here

**Comment by tklauser** [comment]
@tonistiigi @thaJeztah Thanks! Rebased and dropped the vendoring changes.

**Comment by thaJeztah** [comment]
Thanks!

Hm.. interesting; we enabled the new `DCO` bot, but it looks like _two_ new DCO-checks were added (there's now also a `docker/dco-signed` check) Perhaps that one already existed but not configured properly, and now got activated as well? 😂 

Screenshot 2019-08-22 at 10 38 36

^^ @tiborvass

**Review by tonistiigi — APPROVED** [review]

File: pkg/system/utimes_unix.go [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
 
 // LUtimesNano is used to change access and modification time of the specified path.
 // It's used for symbol link file because unix.UtimesNano doesn't support a NOFOLLOW flag atm.
 func LUtimesNano(path string, ts []syscall.Timespec) error {
-	var _path *byte
-	_path, err := unix.BytePtrFromString(path)
-	if err != nil {
-		return err
+	uts := []unix.Timespec{
+		unix.NsecToTimespec(syscall.TimespecToNsec(ts[0])),
```
Comment (tiborvass) [inline pkg/system/utimes_unix.go]: might as well change LUtimesNano callers to send a `[]unix.Timespec` instead. Can be done in a followup.

**Review by tiborvass — APPROVED** [review]

File: pkg/system/utimes_unix.go (hunk 1 above)
Comment (tklauser) [reply pkg/system/utimes_unix.go]: Thanks, will send a followup PR.

**Pull request merged**